actionpack/lib/action_dispatch/railtie.rb

```diff
@@ -42,9 +42,11 @@ class Railtie < Rails::Railtie # :nodoc:
       ActionDispatch::Http::URL.tld_length = app.config.action_dispatch.tld_length
       ActionDispatch::Request.ignore_accept_header = app.config.action_dispatch.ignore_accept_header
       ActionDispatch::Request::Utils.perform_deep_munge = app.config.action_dispatch.perform_deep_munge
-      ActionDispatch::Response.default_charset = app.config.action_dispatch.default_charset || app.config.encoding
-      ActionDispatch::Response.default_headers = app.config.action_dispatch.default_headers
-      ActionDispatch::Response.return_only_media_type_on_content_type = app.config.action_dispatch.return_only_media_type_on_content_type
+      ActiveSupport.on_load(:action_dispatch_response) do
+        ActionDispatch::Response.default_charset = app.config.action_dispatch.default_charset || app.config.encoding
+        ActionDispatch::Response.default_headers = app.config.action_dispatch.default_headers
+        ActionDispatch::Response.return_only_media_type_on_content_type = app.config.action_dispatch.return_only_media_type_on_content_type
```

Comment:
This block will run in the context of `ActionDispatch::Response`, and it's conventional to use `self` to refer to it:

https://guides.rubyonrails.org/v6.0.0/engines.html#example-3

```suggestion
        self.return_only_media_type_on_content_type = app.config.action_dispatch.return_only_media_type_on_content_type
```

Reply:
Changed!